packages/core/src/bundle.ts

```diff
@@ -304,7 +304,8 @@ function makeBundleVisitor(
     } else {
       // @ts-ignore
       delete ref.$ref;
-      Object.assign(ref, resolved.node);
+      const obj = Object.assign({}, resolved.node, ref);
+      Object.assign(ref, obj);
```

Comment:
```suggestion
      Object.assign(ref, resolved.node, ref); // assign ref itself again so ref fields take precedence
```

Reply:
It's wrong, because we should get first val info from resolved.node.

Reply:
It's the same. I don't see how it's different. Can you show me some counter-example?